streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -523,7 +523,17 @@ private RepartitionTopics prepareRepartitionTopics(final Cluster metadata) {
         final boolean isMissingInputTopics = !repartitionTopics.missingSourceTopicExceptions().isEmpty();
         if (isMissingInputTopics) {
             if (!taskManager.topologyMetadata().hasNamedTopologies()) {
-                throw new MissingSourceTopicException("Missing source topics.");
```

Comment:
In addition to throwing the exception you would also log the error.

Reply:
Was Bruno meant was, that we need to add `log.error(...)` to log the error message before throwing the exception. Seems you did not add this yet?

Reply:
Adding to a previous comment from Bruno, I am wondering if we should also change the error log in `StreamsRebalanceListener` to point out that the missing source topic names might be logged on a different instance?

The `StreamsRebalanceListener` is executed on every instance, but `StreamsPartitionAssignor` only on the group leader.

Reply:
@mjsax thanks for your comments! 
> Was Bruno meant was, that we need to add log.error(...) to log the error message before throwing the exception. Seems you did not add this yet?

No, i created new commits. you can see image below.
In this Image, Bruno say `In addition to throwing the exception you would also log the error.`
<img width="782" alt="image" src="https://github.com/apache/kafka/assets/90125071/dc652778-ec5d-4d2b-90d4-950839adf2aa">

and then, I have made these changes to reflect that comment to maintain readability.
you can see this [commit](https://github.com/apache/kafka/commit/3f16d28fd07bec76a980b184d7ba708d7d7c9b0c#diff-571894e60232d30f3754d03d122d653e5f2cff4fa646a59ba25688f9aea77318R526-R527). 
<img width="802" alt="image" src="https://github.com/apache/kafka/assets/90125071/f2551117-a9a0-4b75-8886-063aa0dc6b34">

Reply:
<img width="948" alt="image" src="https://github.com/apache/kafka/assets/90125071/c00ef63e-c20b-4edf-9795-7b8b2b314975">
I draw image above to describe `idea for improve`. 
If you want to read `MissingSourceTopics` in StreamRebalanceListener as well, this workaround seems like it could be a way to handle it.

1. As you know, `ThreadLocal` provides storage specific to each thread.
2. Both `StreamThread`, `StreamRebalanceListener`, `StreamPartitionAssignor` are included on internal package. that means, it is not public API.
3. `StreamRebalanceListener` has reference of `StreamThread` Already. 

Thus, all the things that we should do, are 3step.
1. Add `ThreadLocal` to field of `StreamThread`.
2. Add method that put `missing source topics` to `ThreadLocal` before throw `MissingSourceTopicExceptions`, 
3. Add some codes on `StreamRebalanceListener` to get `MissingSourceTopics` from `ThreadLocal`.

Does it make sense to you?

Reply:
Seems there is some misunderstanding. Sorry for causing confusion.

1) This current code is not producing a log message:
```
final String errorMsg = String.format("Missing source topics. %s", repartitionTopics.missingSourceTopics());
throw new MissingSourceTopicException(errorMsg);
```
The code should be something like:
```
final String errorMsg = String.format("Missing source topics. %s", repartitionTopics.missingSourceTopics());
log.error(errorMsg);
throw new MissingSourceTopicException(errorMsg);
```

2) I did not propose to include the topic names... As Bruno already pointed out, it would require a protocol change what seems to be overkill. In `StreamsRebalanceListener`, we currently log 
```
Received error code 1
```
Cf https://github.com/apache/kafka/blob/trunk/streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsRebalanceListener.java#L56

I would propose to actually change `AssignorError` to contain a proper String. Error code `1` does not mean anything to users. Additionally, it might be good to just change the error message of the log line and the exception to say something like: "To check which topics are missing, please look into the logs of the consumer group leader. Only the leaders knows and logs the name of the missing topics."

Reply:
@mjsax Thanks for your kindful descriptions. it was very helpful for me 🙇‍♂️ 

> 1.This current code is not producing a log message:
I misunderstood, thanks for your time.
I added `log.error()` through this [commit](https://github.com/apache/kafka/pull/15573/commits/fb957dbfc757d62fc5fe4ede4921ec011be96fa2).


> I would propose to actually change AssignorError to contain a proper String.
I agree with you. 
Here's how I've implemented it based on your suggestion. 
I don't know all the meaning of error codes exactly, so I referred to the existing error logs message related with error codes to write the messages. 

What do you think? Do you think i'm on right direction?
When you have some free time, Please take a look 🙇‍♂️.